### ISSUE TITLE
Keyboard shortcuts for formatting markdown

### ISSUE THREAD

**Issue by kshitij10496** [fabiospampinato/notable#153]
Thanks for the wonderful app @fabiospampinato  ❤️ 

Would it be possible to add keyboard bindings for formatting markdown texts in notes in a similar way to GitHub text boxes?

- `Cmd` + `b` for emboldening (****)
- `Cmd` + `i` for italicizing (**)

Currently, the user have to manually type all the markdown formatting. 
I believe providing shortcuts would be really useful!

**Comment by fabiospampinato** [comment]
Agreed, doing stuff via shortcuts and avoiding cluttering the UI is the way to go.

I'll check if those shortcuts clash with anything else.

Perhaps there's a plugin for codemirror that does this already.

**Comment by ymjonline** [comment]
> Thanks for the wonderful app @fabiospampinato ❤️
> 
> Would it be possible to add keyboard bindings for formatting markdown texts in notes in a similar way to GitHub text boxes?
> 
> * `Cmd` + `b` for emboldening (****)
> * `Cmd` + `i` for italicizing (**)
> 
> Currently, the user have to manually type all the markdown formatting.
> I believe providing shortcuts would be really useful!

true. Now I would advise using an external editor for your heavy-lifting editing, such as Typora (or MarkText) that are awesome MD editors.
and use Notable mainly for viewing or quick editing - and the management layer around it (tags, notebooks etc)

**Comment by kaufmann42** [comment]
Did this ever get any traction? I think this would be pretty powerful. I followed the references but it seems everything got closed with no reference to any PR or active issue.

**Comment by fabiospampinato** [comment]
@kaufmann42 It seems to be getting some traction in the sense that people are adding their 👍 to this issue. At some point is going to get implemented, but I think we should implement customizable shortcuts #179 first, and if plugins #128 are implement first then anybody could write a plugin for this.

**Comment by fabiospampinato** [comment]
@francescomucio linked to GitHub shortcuts reference: https://help.github.com/en/articles/using-keyboard-shortcuts#comments perhaps we should model our shortcuts for formatting Markdown after that.

**Comment by Bluesir9** [comment]
Hi, I would like to work on this. Are we still waiting for [customizable shortcuts](https://github.com/notable/notable/issues/477) before implementing this or can I get started right away? Or is there anybody else already working on this?

**Comment by fabiospampinato** [comment]
@Bluesir9 There aren't customizable shortcuts yet, since that issue is still open, but if you'd like to get started on this feel free to do so. It would be valuable to have a good list of shortcuts (i.e. one in which the best shortcuts possible are used) and their implementations once I'm ready to implement/merge this.

**Comment by Bluesir9** [comment]
@fabiospampinato Cool, I will get started on implementing the shortcuts used by GitHub mentioned [here](https://help.github.com/en/articles/keyboard-shortcuts#comments).

**Comment by fabiospampinato** [comment]
@Bluesir9 Ok 👍 `Cmd+K` won't be usable though, as there are going to be a bunch of built-in chorded shortcuts that start with that, e.g. `Cmd+K Cmd+B`.

**Comment by orionrush** [comment]
+21 clams for formatting shortcuts. . .

**Comment by Bluesir9** [comment]
Hi, I said I would work on this and I did start but I never got around finishing it. You can find the changes I made in [this](https://github.com/Bluesir9/notable/tree/add_more_shortcuts) branch. I can't comment on how well-written it was but functionally it worked as expected. If anyone want's to pickup where I left off then feel free. 

Very sorry for leaving this hanging around without any progress for so long. 😐
